crates/sui-config/src/lib.rs

```diff
@@ -0,0 +1,501 @@
+    committee_config: CommitteeConfig,
+}
+
+impl Config for ValidatorConfig {}
+
+impl ValidatorConfig {
+    pub fn key_pair(&self) -> &KeyPair {
+        &self.key_pair
+    }
+
+    pub fn public_key(&self) -> PublicKeyBytes {
+        *self.key_pair.public_key_bytes()
+    }
+
+    pub fn sui_address(&self) -> SuiAddress {
+        SuiAddress::from(self.public_key())
+    }
+
+    pub fn db_path(&self) -> &Path {
+        &self.db_path
+    }
+
+    pub fn network_address(&self) -> &Multiaddr {
+        &self.network_address
+    }
+
+    pub fn consensus_config(&self) -> &ConsensuseConfig {
+        &self.consensus_config
+    }
+
+    pub fn committee_config(&self) -> &CommitteeConfig {
+        &self.committee_config
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct ConsensuseConfig {
+    consensus_address: Multiaddr,
+    consensus_db_path: PathBuf,
+
+    //TODO make narwhal config serializable
+    #[serde(skip_serializing)]
+    #[serde(default)]
+    narwhal_config: DebugIgnore<ConsensusParameters>,
+}
+
+impl ConsensuseConfig {
+    pub fn address(&self) -> &Multiaddr {
+        &self.consensus_address
+    }
+
+    pub fn db_path(&self) -> &Path {
+        &self.consensus_db_path
+    }
+
+    pub fn narwhal_config(&self) -> &ConsensusParameters {
+        &self.narwhal_config
+    }
+}
+
+//TODO get this information from on-chain + some way to do network discovery
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct CommitteeConfig {
+    epoch: EpochId,
+    validator_set: Vec<ValidatorInfo>,
+    consensus_committee: DebugIgnore<ConsensusCommittee<Ed25519PublicKey>>,
+}
+
+impl CommitteeConfig {
+    pub fn epoch(&self) -> EpochId {
+        self.epoch
+    }
+
+    pub fn validator_set(&self) -> &[ValidatorInfo] {
+        &self.validator_set
+    }
+
+    pub fn narwhal_committee(&self) -> &ConsensusCommittee<Ed25519PublicKey> {
+        &self.consensus_committee
+    }
+
+    pub fn committee(&self) -> Committee {
+        let voting_rights = self
+            .validator_set()
+            .iter()
+            .map(|validator| (validator.public_key(), validator.stake()))
+            .collect();
+        Committee::new(self.epoch(), voting_rights)
+    }
+}
+
+/// Publicly known information about a validator
+/// TODO read most of this from on-chain
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct ValidatorInfo {
+    public_key: PublicKeyBytes,
+    stake: usize,
+    network_address: Multiaddr,
+}
+
+impl ValidatorInfo {
+    pub fn sui_address(&self) -> SuiAddress {
+        SuiAddress::from(self.public_key())
+    }
+
+    pub fn public_key(&self) -> PublicKeyBytes {
+        self.public_key
+    }
+
+    pub fn stake(&self) -> usize {
+        self.stake
+    }
+
+    pub fn network_address(&self) -> &Multiaddr {
+        &self.network_address
+    }
+}
+
+/// This is a config that is used for testing or local use as it contains the config and keys for
+/// all validators
```

Comment:
Today's workflow:
1. we collect validator information in `genesis.conf`, socialize it with all genesis validators (there's a keypair which is specific to the concerned validator)
2. each validator runs `sui genesis` and get a `network.conf`. The `network.conf` only has public keys of all validators (plus a keypair of the concerned validator)

Reply:
The new workflow should be more straight forward:
1. We perform genesis and generate a NetworkConfig which includes N Validator configs
2. We give the corresponding validator config to each validator